d2compiler/compile.go

```diff
@@ -735,8 +735,20 @@ func (c *compiler) validateNear(g *d2graph.Graph) {
 					c.errorf(obj.Attributes.NearKey, "constant near keys can only be set on root level shapes")
 					continue
 				}
-				if len(obj.ChildrenArray) > 0 {
-					c.errorf(obj.Attributes.NearKey, "constant near keys cannot be set on shapes with children")
+
+				var edgeConnectToOutsideOfNear *d2graph.Edge
+				for _, edge := range g.Edges {
```

Comment:
this loop should be outside the surrounding loop.

Reply:
fixed, the new logic also need to validae it's `isConstant` or `isKey`